index.js

```diff
@@ -246,7 +253,12 @@ function actualWrite (sonic) {
   sonic._writing = true
   var buf = sonic._buf
   var release = sonic.release
-  sonic._buf = ''
+  if (buf.length > MAX_WRITE) {
+    buf = buf.slice(0, MAX_WRITE)
+    sonic._buf = sonic._buf.slice(MAX_WRITE)
+  } else {
+    sonic._buf = ''
+  }
```

Comment:
I don't think this will be an issue but I just want to raise the it so we can eliminate it:

Are there any scenarios where splitting utf8 chars over this boundary will become an issue?

Reply:
`slice`  is utf-8 aware, so a multi-byte char is going to be counted as 1. This is not a problem for this estimate.

Reply:
It does't seem to be utf8 aware? seems to slice exactly on the byte boundary:
![image](https://user-images.githubusercontent.com/1190716/61656465-1b685a00-acc1-11e9-8e01-9ded1a2e4c53.png)

Reply:
'does not seem to be?😮'.substring(21)